app/code/Magento/CheckoutAgreements/Test/Unit/Model/Checkout/Plugin/GuestValidationTest.php

```diff
@@ -150,4 +185,44 @@ public function testBeforeSavePaymentInformation()
             $this->addressMock
         );
     }
+
+    /**
+     * Build mock object for search criteria
+     *
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    private function buildSearchCriteria() : \PHPUnit_Framework_MockObject_MockObject
+    {
+        $storeId = 1;
+        $storeMock = $this->createMock(\Magento\Store\Model\Store::class);
+        $storeMock->expects($this->any())->method('getId')->will($this->returnValue($storeId));
+        $this->storeManagerMock->expects($this->any())->method('getStore')->will($this->returnValue($storeMock));
+
+        $storeFilterMock = $this->createMock(\Magento\Framework\Api\Filter::class);
+        $activeFilterMock = $this->createMock(\Magento\Framework\Api\Filter::class);
+
+        $this->filterBuilderMock->expects($this->at(0))->method('setField')->with('store_id')->willReturnSelf();
```

Comment:
This kind of test looks like it is intended to freeze code state but not ensure quality.
I would suggest using other approaches or even test types.

Reply:
This logic also covered with integration test \Magento\CheckoutAgreements\Model\Checkout\Plugin\GuestValidationTest

Reply:
This unit test may be redundant in this case. You may remove it or leave as is.